File: 1.1 Modular Programming/example_code_R/case_study/answers/joins.r


#' Join the required frames on specified columns, 
#' dropping unecessary columns
join_frames <- function(left_dataframe, right_dataframe, left_column, right_column) {
  
  # Change location_id to be called country_code for join
  colnames(right_dataframe)[colnames(right_dataframe) == right_column] <- left_column
  
  combined_frames <- dplyr::left_join(x = left_dataframe,
                                      y = right_dataframe,
                                      by = left_column)
  
  combined_frames_reduced <- dplyr::select(combined_frames, sdg_region_name, population_density)
  
  return(combined_frames_reduced)
}

#' Join a list of frames with an inner join on a specified column name
join_years <- function(dataframes, join_column) {

  merged_frame <- dataframes %>% purrr::reduce(.x = dataframes, 
                                                .f = dplyr::inner_join, 
                                                by = join_column)

  return(merged_frame)
}

File: 4. Reproducible Code/mod_prog_py_r/course_content/example_code_python/case_study/answers/joins.py
import pandas as pd


def join_frames(left_dataframe, right_dataframe, left_column, right_column):
    """
    Function to join the required frames on specified columns, dropping
    unrecessary column
    """
    combined_frames = left_dataframe.merge(right=right_dataframe,
                                           how="left",
                                           left_on=left_column,
                                           right_on=right_column)
                                           
    combined_frames_reduced = combined_frames.drop(labels=[right_column], axis=1)
    
    return combined_frames_reduced
    
    
    
def join_years(dataframes, join_column):
    """Function to join a list of frames with an inner join on a specified column name"""
    
    # Starting with first frame iterate over each other frame and join to initial
    temp_frame = dataframes[0]
    
    for frame in dataframes[1:]:
        temp_frame = temp_frame.merge(frame, how="inner", on=join_column)
    
  
    return temp_frame
